src/errors.ts

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2015-2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export class InvalidStoreError extends Error {
+    static TOGGLED_LAZY_LOADING = "TOGGLED_LAZY_LOADING";
+    reason: string;
+    value: unknown;
+
+    constructor(reason: string, value: unknown) {
+        const message = `Store is invalid because ${reason}, `
+            + `please stop the client, delete all data and start the client again`;
+        super(message);
+        this.reason = reason;
+        this.value = value;
```

Comment:
we should use the shortcuts in the constructor rather than this

Reply:
Can't do that as there seems to be a bug in typescript which isn't allowing even statements like variable assignment (`const message = ...`) before the `super()` call.
<img width="634" alt="image" src="https://user-images.githubusercontent.com/19413217/150947305-5f330856-5842-4841-a58e-aebb9b45170c.png">
[Bug reference](https://github.com/microsoft/TypeScript/issues/3311)

Reply:
Why not inline the variable into the super call then?

Reply:
Done, although imo this way makes it more unclear, since the parameter isn't a small simple string.

Also, shouldn't these rules be enforced using eslint? Else these small coding style issues will keep on appearing in future pull requests too.